rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildExecutionEndpoint.java

```
@@ -0,0 +1,148 @@
+            if(loggedUser != null && loggedUser != "") {
+                currentUser = datastore.retrieveUserByUsername(loggedUser);
+            } else {
+                return Response.status(Response.Status.FORBIDDEN).build();
+            }
+
+            final BuildConfiguration configuration = buildConfigurationRepository.queryById(buildConfigurationId);
+            IdRev idRev = new IdRev(buildConfigurationId, buildConfigurationRevision);
+            final BuildConfigurationAudited configurationAudited = buildConfigurationAuditedRepository.queryById(idRev);
+
+            Consumer<BuildStatus> onComplete = (buildStatus) -> {
+                if (callbackUrl == null || callbackUrl.isEmpty()) {
+                    // Expecting URL like: http://host:port/business-central/rest/runtime/org.test:Test1:1.0/process/instance/7/signal?signal=testSig
+                    bpmCallback.signalBpmEvent(callbackUrl.toString() + "&event=" + buildStatus);
+                }
+            };
+
+            Set<Integer> buildRecordSetIds = parseIntegers(buildRecordSetIdsCSV);
+            BuildExecutionTask buildExecutionTask = buildExecutor.build(
+                    configuration,
+                    configurationAudited,
+                    currentUser,
+                    onComplete,
+                    buildRecordSetIds,
+                    Integer.parseInt(buildConfigSetRecordId),
+                    buildTaskId);
+
+            UriBuilder uriBuilder = UriBuilder.fromUri(uriInfo.getBaseUri()).path("/result/running/{id}");
+            int runningBuildId = buildExecutionTask.getId();
+            URI uri = uriBuilder.build(runningBuildId);
+            return Response.ok(uri).header("location", uri).entity(buildRecordProvider.getSpecificRunning(runningBuildId)).build();
+        } catch (BuildConflictException e) {
+            return Response.status(Response.Status.CONFLICT).entity(buildRecordProvider.getSpecificRunning(e.getBuildTaskId())).build();
```

You mean to use throw instead of return ?